src/components/MdComponents.tsx

```diff
@@ -43,40 +43,39 @@ import InfoBanner from "./InfoBanner"
 /**
  * Base HTML elements
  */
-const headingPropsForAnchor = (id?: string): HeadingProps =>
-  id
-    ? ({
-        scrollMarginTop: 28,
-        id,
-        "data-group": true,
-        position: "relative",
-      } as HeadingProps)
-    : {}
+const headingPropsForAnchor = (id?: string): HeadingProps => {
+  if (!id) return {}
+  return {
+    scrollMarginTop: 28,
+    id,
+    "data-group": true,
+    position: "relative",
+  } as HeadingProps
+}
 
 export const commonHeadingProps = (id?: string): HeadingProps => ({
   fontWeight: 700,
   lineHeight: 1.4,
   ...headingPropsForAnchor(id),
 })
 
-const IdAnchor: React.FC<{ id?: string }> = ({ id }) =>
-  id ? (
+const IdAnchor: React.FC<{ id?: string }> = ({ id }) => {
+  if (!id) return null
+  return (
     <Link
-      href={`#${id}`}
+      href={"#" + id}
       position="absolute"
       insetInlineEnd="100%"
       aria-label={id.replaceAll("-", " ") + " permalink"}
+      opacity={0}
+      _groupHover={{ opacity: 1 }}
+      _focus={{ opacity: 1 }}
+      transition="opacity 0.1s ease-in-out"
```

Comment:
Moved this into parent `Link` which takes the `_focus` state.

Above updates are just small clean up items